nestkernel/connection_manager.cpp

```diff
@@ -1087,8 +1087,8 @@ nest::ConnectionManager::get_connections( DictionaryDatum params ) const
 
 // Helper method, implemented as operator<<(), that removes ConnectionIDs from
 // input deque and appends them to output deque.
-static inline std::deque< nest::ConnectionID >& operator<<(
-  std::deque< nest::ConnectionID >& out,
+static inline std::deque< nest::ConnectionID >&
+join_deques( std::deque< nest::ConnectionID >& out,
```

Comment:
The `static` is confusing here, since this use of `static` is so rare hardly anyone knows what it does. It does make sense, as it makes sure there won't be any conflicts with functions of same name and signature defined in other cpp-files, but you should probably comment on this here.

I am also wondering a little about the name. Maybe `extend( connectome, conns_in_thread )` or `add_to( connectome, conns_in_thread )` would make for even more readable code.